File: Python/2391. Minimum Amount of Time to Collect Garbage.py
class Solution:
    def garbageCollection(self, garbage: List[str], travel: List[int]) -> int:
        m, p, g = 0, 0, 0
        for i in range(len(garbage)):
            temp = garbage[i]
            if "M" in temp: m = i
            if "P" in temp: g = i
            if "G" in temp: p = i

        return sum(travel[:m]) + sum(travel[:p]) + sum(travel[:g]) + len(list(''.join(garbage)))

File: Java/2391. Minimum Amount of Time to Collect Garbage.java
class Solution {
    public int garbageCollection(String[] garbage, int[] travel) {
        int m = 0, p = 0, g = 0, l = 0;
        String temp;

        for(int i = 0; i < garbage.length; i++){
            temp = garbage[i];
            if(temp.indexOf("M") != -1) m = i;
            if(temp.indexOf("P") != -1) p = i;
            if(temp.indexOf("G") != -1) g = i;
            l += temp.length();
        }

        int res = 0, t = 0;
        for(int i = 0; i < travel.length; i++){
            t = travel[i];
            if(m > 0){
                res += t;
                m--;
            }
            if(p > 0){
                res += t;
                p--;
            }
            if(g > 0){
                res += t;
                g--;
            }
        }

        return res + l;
    }
}
